### DISCUSSION TITLE
Conditionally subscribe to a notification

### DISCUSSION THREAD

**Discussion by sebastianwhiffen** [LuckyPennySoftware/MediatR#1031] [Q&A]
Hey, is there any existing way to conditionally subscribe to a class implementing : INotification ?

the .Publish method is nice but depending on my projects use I may want to exclude some events (or more specifically not include them in the first place) in different scenarios

is this possible to configure on startup or am I stuck without resorting to using something like [https://github.com/StevenRasmussen/MediatR.Dynamic](https://github.com/StevenRasmussen/MediatR.Dynamic)

Thanks in advance for any replies

**Comment by GabrieleTronchin** [comment]
Hi @sebastianwhiffen,

Try creating another interface that inherits from INotification, for example, IMyNewNotification : INotification. 
In this way, you can subscribe only to messages that implement the custom interface.
Similarly, you can create pipelines that intercept only this family of notifications, rather than all notifications.

As a general suggestion, avoid using the plain IMediator interface directly.
